pkg/operation/botanist/matchers/matcher.go

```diff
@@ -60,6 +60,14 @@ var (
 		v1beta1constants.ShootNoCleanup: "true",
 		managedresources.LabelKeyOrigin: managedresources.LabelValueGardener,
 	}
+
+	// WebhookConstraintMatchersForLeases contains a list of lease api resources which can break
+	// the waking up of a cluster.
+	WebhookConstraintMatchersForLeases = []WebhookConstraintMatcher{
+		{GVR: coordinationv1.SchemeGroupVersion.WithResource("leases"), NamespaceLabels: kubeSystemNamespaceLabels},
+		{GVR: coordinationv1beta1.SchemeGroupVersion.WithResource("leases"), NamespaceLabels: kubeSystemNamespaceLabels},
+	}
```

Comment:
Instead of introducing this, can't we extend the `WebhookConstraintMatcher` with a new attribute which specifies the tolerable timeout seconds? Then we can implement it more generically in case we find a similar scenario tomorrow.

Reply:
To extend `WebhookConstraintMatcher` with a new attribute that specifies the tolerable timeout seconds, `WebhookConstraintMatcher` will also need a field for the failure policy otherwise all the webhook with failure policy `Ignore` and `timeoutSeconds` greater than 15 seconds will also get considered problematic.

IMO current approach is good, if we get such corner cases in future then we can extend `WebhookConstraintMatcher` to also have `timeoutSeconds` and `FailurePolicy`.